mantidproject/mantid
# CMake updates for building with with boost python 2 or 3

This updates the cmake code for finding boost python so that it finds the version appropriate to the version of python being used.

The naming the library linked to python3 is not consistent across platforms. The standard form is `boost_pythonSUFFIX`. A list of known suffixes is used by default and a cmake user variable `BOOST_PYTHON_ADDITIONAL_SUFFIXES` has been defined to allow the user to supply additional ones to check. 

Note: This has had the branch from #16125 merged into it.

**To test:**

Run

```
cmake -DPYTHON_EXECUTABLE=/usr/bin/python3
```

and observe that the correct boost python library has been found.

No issue number.

_Does not need to be in the release notes._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
